util/taskcluster.py

```diff
@@ -113,7 +113,15 @@ def read(fname):
         if plat == 'darwin':
             plat = 'macosx_10_10'
 
-        ds_version = parse_version(read('../VERSION'))
+        version_string = read('../VERSION').strip()
+        ds_version = parse_version(version_string)
+
+        # Try and detect that --branch was not passed and thus "master" is coming
+        # as the default value: we want to force fetch from tag-based
+        # We try to detect '--branch xxx' or '--branch=xxx'
+        has_branch_set = len(filter(lambda x: x == True, map(lambda x: '--branch' == x or '--branch=' in x, sys.argv))) > 0
```

Comment:
I'd rather remove the default value for the `--branch` param and then set the default ourselves if it isn't specified, like we do for `--arch`.